utils/src/hasher.rs

```diff
@@ -0,0 +1,60 @@
+//! This module includes the [CryptoDigest] trait,
```

Comment:
Nit: `includes` is probably not the right word here. `exposes`/`provides`/`defines` might be better suited